bytesconv.go

```diff
@@ -384,14 +384,13 @@ func b2s(b []byte) string {
 //
 // Note it may break if string and/or slice header will change
 // in the future go versions.
-func s2b(s string) []byte {
-	sh := (*reflect.StringHeader)(unsafe.Pointer(&s))
-	bh := reflect.SliceHeader{
-		Data: sh.Data,
-		Len:  sh.Len,
-		Cap:  sh.Len,
-	}
-	return *(*[]byte)(unsafe.Pointer(&bh))
+func s2b(s string) (b []byte) {
+	bh := (*reflect.SliceHeader)(unsafe.Pointer(&b))
+	sh := *(*reflect.StringHeader)(unsafe.Pointer(&s))
```

Comment:
Isn't it better to keep this as a pointer instead of dereferencing it? Now sh will take `sizeof(reflect.StringHeader)` bytes of stack space instead of 8 for just a pointer. Or is dereferencing once faster as you don't have to do it for all 3 assignments below?